core-web/libs/edit-content/src/lib/components/dot-edit-content-sidebar/components/dot-edit-content-sidebar-history/components/dot-history-timeline-item/dot-history-timeline-item.component.ts

```diff
@@ -0,0 +1,138 @@
+import { CommonModule, DatePipe } from '@angular/common';
+import { ChangeDetectionStrategy, Component, computed, input, inject, output } from '@angular/core';
+
+import { AvatarModule } from 'primeng/avatar';
+import { ButtonModule } from 'primeng/button';
+import { ChipModule } from 'primeng/chip';
+import { MenuModule } from 'primeng/menu';
+import { TooltipModule } from 'primeng/tooltip';
+
+import { DotCMSContentletVersion } from '@dotcms/dotcms-models';
+import { DotGravatarDirective, DotMessagePipe, DotRelativeDatePipe } from '@dotcms/ui';
+
+import {
+    DotHistoryTimelineItemAction,
+    DotHistoryTimelineItemActionType
+} from '../../../../../../models/dot-edit-content.model';
+
+/**
+ * Component that displays a single history timeline item with version details and actions.
+ * Shows version information, user details, status chips, and provides action menu.
+ *
+ * @example
+ * ```html
+ * <dot-history-timeline-item
+ *   [item]="versionItem"
+ *   (actionTriggered)="onTimelineItemAction($event)">
+ * </dot-history-timeline-item>
+ * ```
+ */
+@Component({
+    selector: 'dot-history-timeline-item',
+    standalone: true,
+    imports: [
+        CommonModule,
+        AvatarModule,
+        ButtonModule,
+        ChipModule,
+        MenuModule,
+        TooltipModule,
+        DotGravatarDirective,
+        DotMessagePipe,
+        DotRelativeDatePipe
+    ],
+    providers: [DatePipe, DotMessagePipe],
+    templateUrl: './dot-history-timeline-item.component.html',
+    styleUrls: ['./dot-history-timeline-item.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotHistoryTimelineItemComponent {
+    private readonly datePipe = inject(DatePipe);
+    private readonly dotMessagePipe = inject(DotMessagePipe);
+
+    /**
+     * The version item to display
+     * @readonly
+     */
+    $item = input.required<DotCMSContentletVersion>({ alias: 'item' });
+
+    /**
+     * Event emitted when an action is triggered on the timeline item
+     */
+    actionTriggered = output<DotHistoryTimelineItemAction>();
+
+    /**
+     * Computed signal for cached translations map
+     * Ensures reactive updates when language changes
+     */
+    private readonly $labels = computed(() => ({
+        preview: this.dotMessagePipe.transform('edit.content.sidebar.history.menu.preview'),
+        restore: this.dotMessagePipe.transform('edit.content.sidebar.history.menu.restore'),
+        compare: this.dotMessagePipe.transform('edit.content.sidebar.history.menu.compare'),
+        delete: this.dotMessagePipe.transform('edit.content.sidebar.history.menu.delete')
+    }));
```

Comment:
a Computed is triggered by other signal. The service no is a signal.
Just create a normal Signal.